examples/src/new-sdk/producer/main.rs

```diff
@@ -0,0 +1,73 @@
+use clap::Parser;
+use iggy::client_provider;
+use iggy::client_provider::ClientProviderConfig;
+use iggy::clients::client::IggyClient;
+use iggy::clients::producer::IggyProducer;
+use iggy::messages::send_messages::{Message, Partitioning};
+use iggy::utils::duration::IggyDuration;
+use iggy_examples::shared::args::Args;
+use iggy_examples::shared::messages_generator::MessagesGenerator;
+use std::error::Error;
+use std::str::FromStr;
+use std::sync::Arc;
+use tracing::info;
+
+#[tokio::main]
+async fn main() -> anyhow::Result<(), Box<dyn Error>> {
+    let args = Args::parse();
+    tracing_subscriber::fmt::init();
+    info!(
+        "New SDK producer has started, selected transport: {}",
+        args.transport
+    );
+    let client_provider_config = Arc::new(ClientProviderConfig::from_args(args.to_sdk_args())?);
+    let client = client_provider::get_raw_connected_client(client_provider_config).await?;
+    let client = IggyClient::builder().with_client(client).build()?;
+    let mut producer = client
+        .producer(&args.stream_id, &args.topic_id)?
+        .batch_size(args.messages_per_batch)
+        .send_interval(IggyDuration::from_str(&args.interval)?)
+        .partitioning(Partitioning::balanced())
+        .create_topic_if_not_exists(3, None)
```

Comment:
Maybe we could rename this to `bootstrap_resources` and create everything that is required, stream including ? and make the partition count have a default that would be over-writeable throught the `producer` method call

Reply:
These are not required, as they already have the default values in place :)